addons/mail_group/tests/test_mail_group.py

```diff
@@ -110,6 +110,17 @@ def test_find_member_for_alias(self):
         self.test_group.alias_id.alias_contact = 'followers'
         self.assertFalse(self.test_group._alias_get_error_message({}, msg_dict, self.test_group.alias_id))
 
+    def test_find_group_user_for_alias(self):
+        """Check for mail incoming from an allowed group. Specifically for a situation where the sender is a part of the allowed user group, but is NOT a member of the mailing list."""
+        group_user_not_member = mail_new_test_user(self.env, login='group user not member', email="group_user_not_member@example.com")
+
+        self.assertIn(group_user_not_member, self.test_group.access_group_id.users, "User, that sends e-mail, must be part of the access group")
+        self.assertNotIn(group_user_not_member.id, self.test_group.member_ids.ids, "User, that sends e-mail, can't be a member of the mail group")
```

Comment:
"can't" or "should not"? As nle pointed out I don't think it's forbidden, if it's just to ensure the test data is consistent it should be made clear

Reply:
yes, it's only to ensure test data looks as required to perform test. i.e. to "setup" the test.
I modified messages to reflect that